### PR TITLE
Handle Jetpack Connection for non-connected sites

### LINKED ISSUE #7525: [Login] Handle Jetpack connection for self-hosted sites without site-level connection
Jetpack connection have two stages connection:
1. Site level connection that gets setup on the initial setup of Jetpack, at this step, just some Jetpack features are available.
2. Account level connection that gets setup once the user connects the site to their WordPress.com account.

The first part is done for each site just once, while the second part can be done for each account separately, and Jetpack features won't be fully available until the second step is done. (source: PCYsg-kqV-p2).

When working on #7307, we implemented Jetpack Connection, but we found out that our logic was working only for the site's that have already did the step 1, this issue is tracking the implementation for sites without the site-level connection too.

### Techinal details:
To handle the connection, the apps currently call the Jetpack endpoint [`/jetpack/v4/connection/url`](https://github.com/Automattic/jetpack/blob/trunk/docs/rest-api.md#get-wp-jsonjetpackv4connection), this API returns a different response depending on the state of the Jetpack connection on the site:
- If the site didn't implement the site-level connection, the URL would be at the form: `https://{site_url}/wp-admin/admin.php?page=jetpack&action=register&_wpnonce={nonce}`.
- If the site has at least a site-level connection, the URL would directly point to WordPress.com servers: `https://jetpack.wordpress.com/jetpack.authorize/1/...`

The constraint with the first URL, is that to allow the connection, it has to use the same authentication Cookies as the ones used to retrieve it, and when it comes to our apps, this means two approaches:
1. Syncing the Cookies between the API network client and the WebView used for the connection.
2. The alternative way would be to call the connection URL using the same network client as the API, while ensuring automatic redirection is disabled, then we can retrieve the WordPress.com connection URL using the header `Location`.

### UI details
We just need to ensure that the WordPress.com ToS are mentioned in the screen that initiates the connection.

The rest is similar to what was implemented in #7307.

### PR DESCRIPTION
**Pull request by hichamboushaba**
Closes: #7525

Depends on https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2538

### Description
This PR implements the changes required by #7525, which is handling the Jetpack connection for non-connected sites (sites without site-level connection).
I went with the second approach: calling the connection URL using the network client `no-redirect` and grabbing up the `Location` header from the response, this allows auto-sign in to the WordPress.com account to work nicely in the WebView.

### Testing instructions
For all the scenarios below, please prepare a site that's not connected to Jetpack.

##### Case 1: Connect to Jetpack from the post-login discovery
1. Use a WordPress.com that's not connected to any site.
2. Open the app, then login using WordPress.com account.
3. Notice the "No Stores" error screen.
4. Click on "Enter site address"
5. Enter the site address URL, and then the site credentials.
6. Notice the Jetpack Connection screen.
7. Confirm the presence of WordPress.com consent at the bottom of the screen.
8. Click on "Connect Jetpack to your account"
9. Confirm that a WebView is opened, and that you are automatically signed in to your WordPress.com account.
10. Approve the connection.
11. Confirm you are signed in correctly to the site.

##### Case 2: Connect to Jetpack using site credentials
1. Make sure to clear the app's data.
2. Open the app, then sign in using the site address.
4. Enter the site credentials.
5. Notice the Jetpack connection error screen.
6. Click on "Connect Jetpack to your account"
7. Continue the connection on the WebView.
8. Confirm that WordPress.com account confirmation screen is opened after the connection.
9. Enter the password or the magic link for confirmation.
10. Confirm that you are signed in to the app.

### Images/gif

- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/compose/TextExts.kt [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+
+const val URL_ANNOTATION_TAG = "url"
+
+/**
+ * Creates an [AnnotatedString] from the passed String resource.
+ *
+ * Source: https://stackoverflow.com/a/70757314 with some adjustments
+ */
+@Composable
+fun annotatedStringRes(@StringRes stringResId: Int, vararg args: Any): AnnotatedString {
```
Comment (JorgeMucientes) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/compose/TextExts.kt:28]: Nice! No need to change in this PR but we might be able to combine this with `fun Spanned.toAnnotatedString()` from `SpannedTextUtils.kt` as they basically do the same (with slight differences) 
My fault here as I added `SpannedTextUtils.kt` to the wrong package

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/compose/TextExts.kt [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+    val string = stringResource(id = stringResId, args)
```
Comment (JorgeMucientes) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/compose/TextExts.kt:29]: Interesting. I had several issues with this in the past, as apparently `stringResource` [discards any styled markup html](https://stackoverflow.com/a/68549851) information in the string res, so by the moment you invoke `HtmlCompat.fromHtml(string, HtmlCompat.FROM_HTML_MODE_LEGACY)` the styling using HTML is already lost. When applied to a string like this: 
```
    <string name="login_need_help_finding_email"><u>Need help finding the required email?</u></string>
``` 
The `<u>` won't work. But from what I see in your code it's a matter of properly scaping the string resource using `&lt;`. Good to know 👍🏼  I've wasted quite a few time with this sort of things.

**Review by JorgeMucientes — APPROVED** [review]

**Comment by JorgeMucientes** [comment]
Works great @hichamboushaba. Awesome job fine-tuning all these rare scenarios with different Jepack configurations. It is very interesting to see on these PRs how many different situations can happen in the login which I completely was unaware of. 

I'll leave the merging to you once FLuxC changeset is updated and conflicts solved.

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/compose/TextExts.kt (hunk 2 above)
Comment (hichamboushaba) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/compose/TextExts.kt:29]: Yes, this is also mentioned in the official docs: https://developer.android.com/guide/topics/resources/string-resource#StylingWithHTML.

But for our case, we need to escape them for another reason, it's needed for Glotpress compatibility, if the tags are not escaped, Glotpress doesn't handle it (check https://github.com/woocommerce/woocommerce-android/pull/4116#discussion_r644051502 and paqN3M-3Z-p2), and it seems this is impacting two strings in the app `what_is_wordpress_link` and `login_need_help_finding_email`, which is causing them to be not translated.

Comment (JorgeMucientes) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/compose/TextExts.kt:29]: Oh! TIL about glotpress not suporting the tags. I'll fix those 2 strings asap in my upcoming PR.

**Pull request merged**